src/50_Samples_%26_Templates/Autosuggest.html

```diff
@@ -0,0 +1,306 @@
+<!--
+## Styles
+
+Only a few styles are necessary to implement the autosuggest overlay.
+-->
+  <style amp-custom>
+  /* --------------------------------
+     Necessary styles for the example
+     -------------------------------- */
+  /*
+   * Prevent the autosuggest box from
+   * flowing outside its parent
+   */
+  .autosuggest-container {
+    position: relative;
+  }
+
+  /*
+   * Position the autosuggest box as an
+   * overlay over the form fields.
+   */
+  .autosuggest-box {
+    position: absolute;
+    width: 100%;
+    /* make the overlay opaque */
+    background-color: #fafafa;
+  }
+
+  /*
+   * Prevents a solid outline from appearing
+   * around an item when the autosuggest box
+   * is opened with an item already selected
+   */
+  .select-option.no-outline[selected] {
+    outline: initial;
+  }
+
+  .hidden {
+    display: none;
+  }
+
+  /* ---------------------------------
+     Additional non-autosuggest styles
+     --------------------------------- */
+  .wrapper {
+    padding: 0 15px;
+  }
+
+  .form-grid {
+    width: 100%;
+    display: grid;
+    grid-template-areas:
+      "firstName lastName"
+      "addressLine1 addressLine1"
+      "addressLine2 addressLine2"
+      "cityState cityState"
+      "suggest suggest"
+      "zipCode none"
+      "submit submit";
+  }
+  .first-name { grid-area: firstName; }
+  .last-name { grid-area: lastName; }
+  .address-line-1 { grid-area: addressLine1; }
+  .address-line-2 { grid-area: addressLine2; }
+  .city-state { grid-area: cityState; }
+  .suggest { grid-area: suggest; }
+  .zip-code { grid-area: zipCode; }
+  .submit { grid-area: submit; }
+
+  .form-item {
+    display: flex;
+    flex-flow: column nowrap;
+    margin-top: 4px;
+  }
+
+  .form-item label,
+  .form-item label input {
+    flex: 1 0 0;
+  }
+
+  .search-submit, .select-option {
+    font-size: 1.1em;
+  }
+
+  .search-submit {
+    margin: 10px 0;
+    padding: .5em 1em;
+    color: #444;
+    border: 1px solid #999;
+    background-color: #fafafa;
+    text-decoration: none;
+    border-radius: 2px;
+  }
+
+  input {
+    padding: 5px;
+    border: 2px solid lightgray;
+    line-height: 1.5em;
+  }
+
+  .autosuggest-box {
+    box-shadow: 0px 2px 6px rgba(0,0,0,.3);
+  }
+
+  .select-option {
+    box-sizing: border-box;
+    height: 30px;
+    line-height: 30px;
+    padding-left: 10px;
+  }
+
+  .last-name input {
+    border-left: 0;
+  }
+
+  .select-option.empty {
+    text-align: center;
+  }
+  </style>
+</head>
+<body>
+  <!-- ## Autosuggest form field -->
+  <!-- -->
+  <h2 class="p1">Autosuggest</h2>
+
+  <!--
```

Comment:
Could you also mention how you combined amp-form, amp-list and amp-bind to implement the autosuggest?

Reply:
Ah yes that's a good idea

Reply:
Done!